libraries/api-documenter/src/MarkdownPageRenderer.ts

```diff
@@ -0,0 +1,225 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as fsx from 'fs-extra';
+import * as os from 'os';
+import * as path from 'path';
+
+import {
+  IDomPage,
+  DomElement
+} from './SimpleDom';
+
+import { BasePageRenderer } from './BasePageRenderer';
+
+/**
+ * Helper class used by MarkdownPageRenderer
+ */
+class SimpleWriter {
+  private _buffer: string = '';
+
+  public write(s: string): void {
+    this._buffer += s;
+  }
+
+  public writeLine(s: string = ''): void {
+    this._buffer += s + '\n';
+  }
+
+  // Adds a newline if the file pointer is not already at the start of the line
+  public finishLine(): void {
+    if (this._buffer.substr(-1, 1) !== '\n') {
+      this.write('\n');
+    }
+  }
+
+  public toString(): string {
+    return this._buffer;
+  }
+}
+
+interface IRenderContext {
+  writer: SimpleWriter;
+  insideTable: boolean;
+}
+
+/**
+ * Renders API documentation in the Markdown file format.
+ * For more info:  https://en.wikipedia.org/wiki/Markdown
+ */
+export class MarkdownPageRenderer extends BasePageRenderer {
```

Comment:
>MarkdownPageRenderer [](start = 13, length = 20)

I will add unit tests for this class, however I will do it in a separate PR because it requires some plumbing.